xbmc/video/ContextMenus.cpp

```diff
@@ -68,7 +69,12 @@ bool CMarkWatched::IsVisible(const CFileItem& item) const
       return URIUtils::IsPVRRecordingFileOrFolder(item.GetPath());
   }
   else if (!item.HasVideoInfoTag())
+  {
+    const int currentWindow = CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow();
+    if (item.IsVideo() && (currentWindow == WINDOW_VIDEO_NAV))
```

Comment:
No need to make the expensive Servicebroker call if the cheap isvideo call evaluated to false.